addon/v1alpha1/types_addontemplate.go

```diff
@@ -123,19 +123,33 @@ type HubPermissionConfig struct {
 	// +kubebuilder:validation:Enum:=CurrentCluster;SingleNamespace
 	Type HubPermissionsBindingType `json:"type"`
 
-	// RoleRef is an reference to the permission resource. it could be a role or a cluster role,
-	// the user must make sure it exist on the hub cluster.
-	// +kubebuilder:validation:Required
-	RoleRef rbacv1.RoleRef `json:"roleRef"`
+	// CurrentCluster contains the configuration of CurrentCluster type binding.
+	// It is required when the type is CurrentCluster.
+	CurrentCluster *CurrentCluster `json:"currentCluster,omitempty"`
```

Comment:
```suggestion
	CurrentCluster *CurrentClusterBindingConfig `json:"currentCluster,omitempty"`
```

not sure if it would be better to change the struct name to `CurrentClusterBindingConfig` to match the `SingleNamespaceBindingConfig`, or we can change the `SingleNamespaceBindingConfig` to `SingleNamespace`